scribus/scpainter.cpp

```diff
@@ -1955,8 +1962,18 @@ void ScPainter::drawText(QRectF area, QString text, bool filled, int align)
 	double ww = 0;
 	double hh = 0;
 	double r, g, b;
-	QString fontName = m_font.family() + " " + m_font.style();
-	cairo_select_font_face(m_cr, fontName.toLatin1(), CAIRO_FONT_SLANT_NORMAL, CAIRO_FONT_WEIGHT_NORMAL);
+
+	if (m_font.isNone())
+	{
+		const QFont &font = QApplication::font();
+		setFont(PrefsManager::instance()->appPrefs.fontPrefs.AvailFonts.findFont(font.family(), font.styleName()), font.pointSizeF());
+
+	}
+
+	cairo_font_face_t *cairo_face;
+	cairo_face = cairo_ft_font_face_create_for_ft_face (m_font.ftFace(), 0);
```

Comment:
Is this working for you? I had mysterious Cairo drawing issues with it before, if you remember.